crates/router/src/core/payments.rs

```diff
@@ -5927,6 +5929,76 @@ pub async fn payment_external_authentication(
     ))
 }
 
+#[instrument(skip_all)]
+#[cfg(feature = "v2")]
+pub async fn payment_start_redirection(
+    state: SessionState,
+    merchant_account: domain::MerchantAccount,
+    key_store: domain::MerchantKeyStore,
+    req: api_models::payments::PaymentStartRedirectionRequest,
+) -> RouterResponse<serde_json::Value> {
+    let db = &*state.store;
+    let key_manager_state = &(&state).into();
+
+    let storage_scheme = merchant_account.storage_scheme;
+
+    let payment_intent = db
+        .find_payment_intent_by_id(key_manager_state, &req.id, &key_store, storage_scheme)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::PaymentNotFound)?;
+
+    // validate session expiry
+    helpers::authenticate_client_secret(Some(&payment_intent.client_secret), &payment_intent)?;
```

Comment:
we are passing the same client secret here? should we not take the client secret in the api query params?